### ISSUE TITLE
Bug in density.lpp for point on segment end

### ISSUE THREAD

**Issue by rubak** [spatstat/spatstat#117]
Jakob Gulddahl Rasmussen pointed me to a bug in `density.lpp` for a specific pattern. I cooked up the following reproducible example exposing the error:
``` r
library(spatstat)
#> Loading required package: spatstat.data
#> Loading required package: nlme
#> Loading required package: rpart
#> 
#> spatstat 1.61-0.035       (nickname: 'New improved formula') 
#> For an introduction to spatstat, type 'beginner'
X <- ppp(c(0,1), c(0,0), owin())
L <- linnet(X, edges = matrix(1:2, ncol=2))
Y <- lpp(X, L)
# The left end point is OK
d <- density(Y[1], .1)
# The right end point gives an error
density(Y[2], .1)
#> Error in lpp(coordsX, Lfine): Segment index coordinate 'seg' exceeds bounds
```

**Comment by baddstats** [comment]
This is a bug in the C code for `lixellate`.
I will chase it down.

**Comment by baddstats** [comment]
Fixed in `spatstat 1.61-0.036`
